main.cpp

```diff
@@ -47,10 +47,33 @@ struct Point
 template<typename Type>
 struct Wrapper
 {
-    Wrapper(Type&& t) : val(std::move(t)) 
-    { 
-        std::cout << "Wrapper(" << typeid(val).name() << ")" << std::endl; 
+    Wrapper(Type&& t) : val(std::move(t))
+    {
+        std::cout << "Wrapper(" << typeid(val).name() << ")" << std::endl;
+    }
+
+    void print()
+    {
+        std::cout << "Wrapper::print(" << val << ")" << std::endl;
+    }
+
+    const Type& val;
+};

```

Comment:
It does not make sense to store the constructor argument (which is an R-value Reference) in a Const L-Value reference. 
`val` should be a normal member variable. 
- [ ] revise